src/components/picker/picker.tsx

```diff
@@ -252,94 +248,70 @@ export class Picker {
             }
-
-            return (
-                <div
+            cssList.push('dropdown--list');
+            const emptyMessage = (
+                <p
                     style={{
-                        width: `${boundingRect.width}px`,
+                        color: 'var(--lime-light-grey, #{$lime-light-grey})',
+                        'text-align': 'center',
                     }}
-                    class={`
-                        dropdown--list
-                        mdc-elevation-transition
-                        mdc-elevation--z4
-                        mdc-menu-surface
-                        mdc-menu-surface--open
-                    `}
-                    tabindex="-1"
                 >
-                    <p class="empty-result-message">
-                        {this.emptyResultMessage}
-                    </p>
-                </div>
+                    {this.emptyResultMessage}
+                </p>
             );
+            return this.renderPortal(true, emptyMessage, cssList);
         }
 
         const hasIcons = this.items.some(item => {
             return 'icon' in item && !!item.icon;
         });
+        const limelList = (
+            <limel-list
+                badgeIcons={hasIcons}
+                onChange={this.handleListChange}
+                type="selectable"
+                items={this.items}
+            />
+        );
+
+        cssList.push('dropdown--list');
+        if (this.displayFullList || this.items.length <= ITEM_LIMIT_NO_SCROLL) {
+            cssList.push('display-full-list');
+        }
+        return this.renderPortal(true, limelList, cssList);
+    }
 
+    private renderPortal(isVisible, content = null, cssList: string[] = []) {
         return (
-            <div
-                style={{
-                    width: `${boundingRect.width}px`,
-                }}
-                class={`
-                    dropdown--list
-                    mdc-elevation-transition
-                    mdc-elevation--z4
-                    mdc-menu-surface
-                    mdc-menu-surface--open
-                    ${
-                        this.displayFullList ||
-                        this.items.length <= ITEM_LIMIT_NO_SCROLL
-                            ? 'display-full-list'
-                            : ''
-                    }
-                `}
-                tabindex="-1"
-                onKeyDown={this.handleDropdownKeyDown}
-            >
-                <limel-list
-                    badgeIcons={hasIcons}
-                    onChange={this.handleListChange}
-                    type="selectable"
-                    items={this.items}
-                />
-            </div>
+            <limel-portal visible={isVisible} containerId={this.portalId}>
+                <limel-menu-surface
+                    open={isVisible}
+                    cssClasses={cssList}
+                    onDismiss={this.handleListClosed}
```

Comment:
What do you think about naming this `handleMenuClosed`?

Reply:
not sure. it makes sense as a listener on the menu-surface but as a method name in limel-picker it's rather confusing because the picker doesn't really operate with a menu.

Reply:
Maybe `handleSurfaceClosed`? `handleSurfaceDismissed`?